superset-frontend/src/SqlLab/components/App.jsx

```diff
@@ -86,7 +87,13 @@ class App extends React.PureComponent {
 
   render() {
     let content;
-    if (this.state.hash) {
+    if (this.state.hash && this.state.hash === '#search') {
+      if (
+        isFeatureEnabled(FeatureFlag.ENABLE_REACT_CRUD_VIEWS) &&
+        isFeatureEnabled(FeatureFlag.SIP_34_QUERY_SEARCH_UI)
+      ) {
+        return window.location.assign('/superset/sqllab/search');
```

Comment:
does this overwrite the history for this page? Want to make sure that the user can't get stuck in an infinite loop trying to go back from the redirected page

Reply:
Perhaps we can use react router for this too? I don't know what the frontend app structure is like now, but i've found relying on `window` functions to control history can be error prone

Reply:
There's no react-router in this "app", it'll just 404 or complain that no router is preset. This should probably be `window.location.replace`, though. This is only meant to be temporary until this route is fully deprecated, this code is removed, and the backend handles all redirecting to `/superset/sqllab/search`